### PR TITLE
Cover upload_logs in testapi.pm

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/130420

### CODE REVIEW

File: t/03-testapi.t [hunk 1]
```diff
@@ -472,6 +472,70 @@ subtest 'check_assert_screen' => sub {
...
 
+subtest 'upload_logs' => sub {
+    $bmwqemu::vars{AUTOINST_URL_HOSTNAME} = 'localhost';
+    $bmwqemu::vars{QEMUPORT} = '4242';
+    $bmwqemu::vars{JOBTOKEN} = 'LookAtMeImAToken';
+    $cmds = [];
+    upload_logs '/var/log/messages';
+    is_deeply($cmds, [
+            {
+                'text' => 'curl --form upload=@/var/log/messages --form upname=basetest-messages http://localhost:4243/LookAtMeImAToken/uploadlog/messages',
```
Comment (Martchus) [inline t/03-testapi.t:483]: The Perl style check doesn't like the quotes around the keys here:

```
5: ::warning file=/__w/os-autoinst/os-autoinst/t/03-testapi.t,line=483,col=17,title=Hash key 'text' with quotes - severity 4::[HashKeyQuotes] Avoid useless quotes
```

(If it was me we wouldn't enforce this and I think we also don't enforce this in openQA.)

Comment (asdil12) [reply t/03-testapi.t]: I updated it.
In general though, could we either change that or provide a perltidy config that automatically deals with it?
@perlpunk

Comment (kalikiana) [reply t/03-testapi.t]: It doesn't seem to be supported? https://metacpan.org/dist/Perl-Tidy/view/bin/perltidy

**Comment by kalikiana** [comment]
```
3: #   Failed test 'No tests run for subtest "parse_extra_log"'
3: #   at t/03-testapi.t line 556.
3: Can't locate OpenQA/Parser.pm in @INC (you may need to install the OpenQA::Parser module) (@INC entries checked: /usr/share/openqa/lib /opt/blib/arch /opt/blib/lib /opt/t/../external/os-autoinst-common/lib /opt/external/os-autoinst-common/lib /opt/t/lib /opt /opt/ppmclibs/blib/lib /opt/ppmclibs/blib/arch/auto/tinycv /usr/lib/perl5/site_perl/5.40.0/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.40.0 /usr/lib/perl5/vendor_perl/5.40.0/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.40.0 /usr/lib/perl5/5.40.0/x86_64-linux-thread-multi /usr/lib/perl5/5.40.0 /usr/lib/perl5/site_perl) at /usr/lib/perl5/vendor_perl/5.40.0/Test/MockModule.pm line 55.
```

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**